validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java

```diff
@@ -0,0 +1,114 @@
+package com.amazon.aoc.validators;
+
+import com.amazon.aoc.callers.ICaller;
+import com.amazon.aoc.exception.BaseException;
+import com.amazon.aoc.exception.ExceptionCode;
+import com.amazon.aoc.fileconfigs.FileConfig;
+import com.amazon.aoc.helpers.MustacheHelper;
+import com.amazon.aoc.helpers.RetryHelper;
+import com.amazon.aoc.models.Context;
+import com.amazon.aoc.models.ValidationConfig;
+import com.amazon.aoc.services.CloudWatchService;
+import com.amazonaws.services.logs.model.OutputLogEvent;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonschema.main.JsonSchema;
+import com.github.fge.jsonschema.main.JsonSchemaFactory;
+import com.github.fge.jsonschema.report.ListReportProvider;
+import com.github.fge.jsonschema.report.LogLevel;
+import com.github.fge.jsonschema.report.ProcessingReport;
+import com.github.fge.jsonschema.util.JsonLoader;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class CWLogValidator implements IValidator {
+
+  protected String logStreamName = "otlp-logs";
+
+  private static final String LOGGROUPPATH = "/aws/ecs/otlp/%s/logs";
+
+  protected CloudWatchService cloudWatchService;
+  private static final int CHECK_INTERVAL_IN_MILLI = 30 * 1000;
+  private static final int CHECK_DURATION_IN_SECONDS = 2 * 60;
+  private static final int MAX_RETRY_COUNT = 12;
+  private static final int QUERY_LIMIT = 100;
+  private JsonSchema schema;
+  protected String logGroupName;
+
+  private Context context;
+
+  protected final ObjectMapper mapper = new ObjectMapper();
+
+  @Override
+  public void init(
+      Context context,
+      ValidationConfig validationConfig,
+      ICaller caller,
+      FileConfig expectedDataTemplate)
+      throws Exception {
+    this.context = context;
+    cloudWatchService = new CloudWatchService(context.getRegion());
+    logGroupName = String.format(LOGGROUPPATH, context.getTestingId());
+    MustacheHelper mustacheHelper = new MustacheHelper();
+    String templateInput = mustacheHelper.render(expectedDataTemplate, context);
+    JsonNode jsonNode = JsonLoader.fromString(templateInput);
+    JsonSchemaFactory jsonSchemaFactory =
+        JsonSchemaFactory.newBuilder()
+            .setReportProvider(new ListReportProvider(LogLevel.INFO, LogLevel.FATAL))
+            .freeze();
+    JsonSchema jsonSchema = jsonSchemaFactory.getJsonSchema(jsonNode);
+    this.schema = jsonSchema;
+    caller.callSampleApp();
```

Comment:
Why do we call the sample app during init? Should we be calling the sample app each retry attempt instead?

Reply:
Looking at other validators, it seems it is being called in the validate() method, but not in the retry helper(also, I don't think it should be in a retry because calling the sample app uses retry helper on its own right, which would be redundant). I'll move the callSampleApp() to validate() method.